pymagicc/run.py

```diff
@@ -46,6 +46,14 @@ def __init__(self, root_dir=None):
             self.is_temp = True
             self.root_dir = mkdtemp(prefix="pymagicc-")
 
+    def __enter__(self):
+        if not self.is_initialised():
```

Comment:
Then the directory is only initialised if it doesn't already exist, i.e. a temp dir is used, so not to wipe away any existing configuration. Perhaps it should only clean on exit if a new directory was created in the initialiser.

Reply:
So the cases are?

```python
with MAGICC(root_dir="~/MAGICC8-test") as m:
    ...
```

and the other
```python
with MAGICC() as m:
    print(m.is_temp == True)
```

Reply:
Correct (assuming that ~/MAGICC8-test already existed). Do you agree that the clean should only be called if no root_dir was passed?

Reply:
Yes, it's probably clearer to have the check for a temp_dir already in `__exit__`

Reply:
I'm still struggling to see why the check in `__enter__` is done (little experience with context managers ...)

I thought `init` would be called (if no root given) in `__init__` and then I could do something like
```python
out = []

with MAGICC() as m:
    for scen in scenarios:
        out.append(m.run(scen))
```
without creating additional temp copies.

Reply:
After looking at this with @znicholls we figured out the reason of my confusion -- I had thought that 

```python
magicc = MAGICC()
```

would make a copy of the embedded MAGICC resource.
As it is now it's only possible to use the MAGICC class with a context manager or explicitly copying the files.
I think this is right because a user might just want to look at input files etc. which wouldn't require a copy.

Reply:
@lewisjared would it be correct to say that this pull request basically makes it only possible to run MAGICC with a context manager? The only way you could get it to work without is if you pre-copied the directories yourself?

Also can we rename `init` to something less confusing given there's also `__init__`, perhaps `setup`?

Reply:
@znicholls 
It's possible without a context manager:
```python
In [1]: from pymagicc.run import MAGICC

In [2]: magicc = MAGICC()

In [3]: magicc.init()

In [4]: magicc.root_dir
Out[4]: '/tmp/pymagicc-h8zzcs91'
```